pkg/sql/sem/tree/constant.go

```diff
@@ -578,6 +584,11 @@ func (expr *StrVal) ResolveAsType(
 			expr.resString = DString(strings.TrimRight(expr.s, " "))
 			return &expr.resString, nil
 		}
+		// "char" is supposed to truncate long values
+		if typ.Oid() == oid.T_char {
```

Comment:
this logic is very similar to above and the code in `AdjustValueToType`.

Do you mind making a routine
```
func adjustStringToType(typ *types.T, str string) string {
   switch typ.Oid() {
      case oid.T_char:
         // ...
      case oid.T_bpchar:
         // ...
   }
   return str
}
```

and using it in all three places, replacing the TrimRight && TruncateString code?

Reply:
done